src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/message/MergeService.kt

```diff
@@ -19,27 +20,28 @@ import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
 import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.CPR_MERGE_RECORD_NOT_FOUND
 import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.CPR_RECORD_MERGED
 
-@Service
+@Component
 class MergeService(
   private val telemetryService: TelemetryService,
   private val personRepository: PersonRepository,
+  private val personKeyRepository: PersonKeyRepository,
   private val eventLoggingService: EventLoggingService,
   @Value("\${retry.delay}") private val retryDelay: Long,
 ) {
 
-  @Transactional
   fun processMerge(mergeEvent: MergeEvent, sourcePersonCallback: () -> PersonEntity?, targetPersonCallback: () -> PersonEntity?) = runBlocking {
     runWithRetry(MAX_ATTEMPTS, retryDelay, ENTITY_RETRY_EXCEPTIONS) {
       processMergingOfRecords(mergeEvent, sourcePersonCallback, targetPersonCallback)
     }
   }
 
+  @Transactional
   private suspend fun processMergingOfRecords(mergeEvent: MergeEvent, sourcePersonCallback: () -> PersonEntity?, targetPersonCallback: () -> PersonEntity?) {
     val (sourcePersonEntity, targetPersonEntity) = collectPeople(sourcePersonCallback, targetPersonCallback)
 
     when {
-      targetPersonEntity == null -> handleTargetRecordNotFound(mergeEvent)
-      sourcePersonEntity == null -> handleSourceRecordNotFound(mergeEvent, targetPersonEntity)
+      targetPersonEntity == PersonEntity.empty -> handleTargetRecordNotFound(mergeEvent)
```

Comment:
quite nice, could it be done as an extension function on PersonEntity?

Reply:
I wish I knew more Kotlin

Reply:
Yeah i thought about this, but the extension func does not mark the entity as have been null check and still need casting `!!`